### ISSUE TITLE
修改返回为OTSpanContext

### ISSUE THREAD

**Issue by fykyx521** [jcchavezs/zipkin-php-opentracing#46]
https://github.com/jcchavezs/zipkin-php-opentracing/blob/3394cbe5a801816be3baa3c481da70ff0c79a128/src/ZipkinOpenTracing/Tracer.php#L221

SpanContext change to OTSpanContext

**Comment by jcchavezs** [comment]
Could you explain what is the need for this? Would you come up with a PR for this?

**Comment by Rainrider** [comment]
The reason for this is that a [ZipkinOpenTracing\PartialSpanContext](https://github.com/jcchavezs/zipkin-php-opentracing/blob/master/src/ZipkinOpenTracing/PartialSpanContext.php) might be returned, which is not a subtype of [ZipkinOpenTracing\SpanContext](https://github.com/jcchavezs/zipkin-php-opentracing/blob/master/src/ZipkinOpenTracing/SpanContext.php), which is the current return type of this method. As both implement `OpenTracing\SpanContext` maybe this is a more suitable return type in this situation. I would actually expect a boolean return from a method staring with `has`.

I got this from trying out https://github.com/auxmoney/OpentracingBundle-Zipkin:
```
TypeError:
Return value of ZipkinOpenTracing\Tracer::hasParentInOptions() must be an instance of ZipkinOpenTracing\SpanContext or null, instance of ZipkinOpenTracing\PartialSpanContext returned

  at D:\snip\vendor\jcchavezs\zipkin-opentracing\src\ZipkinOpenTracing\Tracer.php:226
  at ZipkinOpenTracing\Tracer->hasParentInOptions(object(StartSpanOptions))
     (D:\snip\vendor\jcchavezs\zipkin-opentracing\src\ZipkinOpenTracing\Tracer.php:86)
  at ZipkinOpenTracing\Tracer->startActiveSpan('https://127.0.0.1:8000/', object(StartSpanOptions))
     (D:\snip\vendor\auxmoney\opentracing-bundle-core\Service\TracingService.php:61)
  at Auxmoney\OpentracingBundle\Service\TracingService->startActiveSpan('https://127.0.0.1:8000/', array('tags' => array('kernel.debug' => 'true', 'kernel.environment' => 'dev', 'symfony.version' => '5.3.7', 'opentracing.version' => 'v0.8.1@c3a26cd7562de6e17f1209079360b0c8a01f2719', 'pod/host' => 'unknown', 'php.version' => '7.4.11', 'http.method' => 'GET', 'http.url' => 'https://127.0.0.1:8000/', 'span.kind' => 'server', 'auxmoney-opentracing-bundle.span-origin' => 'core:request'), 'child_of' => object(PartialSpanContext), 'finish_span_on_close' => true))
     (D:\snip\vendor\auxmoney\opentracing-bundle-core\EventListener\StartRootSpanSubscriber.php:55)
  at Auxmoney\OpentracingBundle\EventListener\StartRootSpanSubscriber->onRequest(object(RequestEvent), 'kernel.request', object(TraceableEventDispatcher))
     (D:\snip\vendor\symfony\event-dispatcher\Debug\WrappedListener.php:117)
  at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(RequestEvent), 'kernel.request', object(TraceableEventDispatcher))
     (D:\snip\vendor\symfony\event-dispatcher\EventDispatcher.php:230)
  at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.request', object(RequestEvent))
     (D:\snip\vendor\symfony\event-dispatcher\EventDispatcher.php:59)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(RequestEvent), 'kernel.request')
     (D:\snip\vendor\symfony\event-dispatcher\Debug\TraceableEventDispatcher.php:151)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(RequestEvent), 'kernel.request')
     (D:\snip\vendor\symfony\http-kernel\HttpKernel.php:132)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (D:\snip\vendor\symfony\http-kernel\HttpKernel.php:78)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (D:\snip\vendor\symfony\http-kernel\Kernel.php:199)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (D:\snip\vendor\symfony\runtime\Runner\Symfony\HttpKernelRunner.php:37)
  at Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner->run()
     (D:\snip\vendor\autoload_runtime.php:35)
  at require_once('D:\\snip\\vendor\\autoload_runtime.php')
     (D:\snip\public\index.php:5)
```

**Comment by jcchavezs** [comment]
Could you please create a failing test case for this?

José Carlos Chávez


tir. 28. sep. 2021 kl. 12:03 skrev Rainrider ***@***.***>:

**Comment by Rainrider** [comment]
I will try this weekend.

**Comment by Rainrider** [comment]
This has been fixed in #49 

The Auxmoney bundle uses an old version of zipkin-php-opentracing. Sorry for loosing your time with this :(

**Comment by jcchavezs** [comment]
No worries. I am happy you came by and also with the fact it is fixed :)
